src/edu/hillel/lesson12/Main.java

```diff
@@ -0,0 +1,138 @@
+package edu.hillel.lesson12;
+
+import java.util.*;
+
+public class Main {
+
+
+    public static void main(String[] args) {
+        //1
+        List<String> strings = new ArrayList<>();
+        strings.add("apple");
+        strings.add("apple");
+        strings.add("apricot");
+        strings.add("lemon");
+        strings.add("apple");
+        strings.add("melon");
+        strings.add("kiwi");
+        strings.add("kiwi");
+        strings.add("citrus");
+        strings.add("avocado");
+        strings.add("avocado");
+        strings.add("watermelon");
+
+        System.out.println("count of kiwi is: "+countOccurrence(strings,"kiwi"));
+        System.out.println("count of apple is: "+countOccurrence(strings,"apple"));
+
+        //2 string
+        String[] stringArray = new String[5];
+        stringArray[0]="apple";
+        stringArray[1]="apricot";
+        stringArray[2]="lemon";
+        stringArray[3]="melon";
+        stringArray[4]="avocado";
+        List<String> stringList = toList(stringArray);
+        System.out.println("New list is: "+stringList);
+        //2 int
+        Integer[] intArray = new Integer[3];
+        intArray[0]=1;
+        intArray[1]=2;
+        intArray[2]=3;
+        List<Integer> intList = toList(intArray);
+        System.out.println("New list is: "+intList);
+
+        //3
+        List<Integer> intList3 = new ArrayList<>();
+        intList3.add(1);
+        intList3.add(2);
+        intList3.add(3);
+        intList3.add(4);
+        intList3.add(1);
+        intList3.add(1);
+        intList3.add(6);
+        intList3.add(4);
+        intList3.add(3);
+        List<Integer> uniqueList = findUnique(intList3);
+        System.out.println("Unique list is: "+uniqueList);
+
+        //4 - calcOccurrence
+        List<String> stringList4 = new ArrayList<>();
+        stringList4.add("bmw");
+        stringList4.add("volvo");
+        stringList4.add("audi");
+        stringList4.add("volvo");
+        stringList4.add("toyota");
+        stringList4.add("bmw");
+        stringList4.add("toyota");
+        stringList4.add("toyota");
+        stringList4.add("zaz");
+        stringList4.add("toyota");
+
+        calcOccurrence(stringList4);
+
+        //4 findOccurrence
+        List<ItemWithCount> items = findOccurrence(stringList4);
+        for (ItemWithCount item : items) {
+            System.out.println("Item name="+item.getName()+", count of occurrence="+item.getCount());
+        }
+    }
+
+    private static int countOccurrence(List<String> strings, String stringToFind){
+        int cnt=0;
+        for (String string : strings) {
+            if (string.compareTo(stringToFind)==0){
```

Comment:
One more question here. Could I ask why did you use compareTo instead of equals?

Reply:
As I understand - equals compare each Object, but for Strings compareTo - comparing with more different ways and can give as more functonality (for example comparing ignore case, compare in different codepages and so on).
Yes - we can change to equals, but compareTo is also possible.

Reply:
I'm used to comparing strings and text....
Ok, I'll change to equal

Reply:
Agree but here more suitable use equals because we don't need exactly know the order of compared string (-1,0,1)